core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -1656,6 +1664,19 @@ def save_info(self) -> dict:
         json_path = self.path + self.model_info_json_name
         save_json.save(path=json_path, obj=info)
         return info
+    
+    def _get_maximum_resources(self) -> Dict[str, Union[int,float]]:
+        """
+        Get the maximum resources allowed to use for this model.
+        This can be useful when model not scale well with resources, i.e. cpu cores.
+        Return empty dict if no maximum resources needed
```

Comment:
This still doesn't tell me the typical keys that are important in the output (aka `num_cpus`, `num_gpus`)